Runtime/AvatarCreator/Data/AvatarTemplateData.cs

```diff
@@ -6,12 +6,13 @@
 namespace ReadyPlayerMe.AvatarCreator
 {
     [Serializable]
-    public class AvatarTemplateData
+    public class AvatarTemplateData : IAssetData
     {
+        public string Id { get; set; }
+        public Category Category { get; set; }
```

Comment:
Maybe if you changed this to be something like an 'AssetType' enum, then it would make more sense, and then you could hard code this to always be of type 'AvatarTemplate' with something like AssetType AssetType => AssetType.AvatarTemplate?

Reply:
Yeah i'm not proud of that 😅
Perhaps AssetType atually does make more sense 🤔 
or alternatively insteead of using an interface and forcing a "categorization" I could have some sort of CustomizationData class (similar to AssetData) and just make a converter to convert from the different types. (TemplateAvatar, PartnerAsset, Colors etc)